lib/v2/components/scanner/scanner_screen.dart

```diff
@@ -95,6 +104,8 @@ class _ScannerScreenState extends State<ScannerScreen> {
         return const CircularProgressIndicator();
       case PageState.error:
         return const SizedBox.shrink();
+      case PageState.stop:
```

Comment:
Refactor a sub-widget tree as constant or method(`Widget buildStateView(state)`) are based on the BuildContext object of the parent widget so each time the parent is redrawn the widget will also do just like with its constants and methods which is not optimal and impacts performace.
I don't think there is a problem with small methods but you always have to look for refactoring in classes. :)

Reply:
https://stackoverflow.com/questions/49713189/how-to-use-conditional-statement-within-child-attribute-of-a-flutter-widget-cen

(() {
  // your code here
}())


is basically a function. Ill keep this one like this. :)

Reply:
Interesting every day I learn something new